src/app/personalised-homepage/page.js

```diff
@@ -0,0 +1,72 @@
+'use client'
+import Aside from "../components/personalised-homepage-components/Aside";
+import ItemCard from "../components/personalised-homepage-components/ItemCard";
+import ItemContainer from "../components/personalised-homepage-components/ItemContainer";
+import ItemRow from "../components/personalised-homepage-components/ItemRow";
+import ItemRowContainer from "../components/personalised-homepage-components/ItemRowContainer";
+import Bg from "../components/personalised-homepage-components/Bg";
+import { useEffect, useState } from "react";
+import { fetchArtistAlbum, fetchProfile, fetchUserPlaylists, fetchUserTracks } from "../hooks/spotify/spotify_hooks";
+
+function PersonalisedHomepage() {
+    let accessToken = "";
+
+    if (typeof window !== "undefined") {
+        accessToken = window.location.hash.split("&")[0].split("=")[1];
+    }
+
+    const [profileName, setProfileName] = useState("");
+    const [userImage, setUserImage] = useState("");
+    const [albums, setAlbums] = useState([]);
+    const [tracks, setTracks] = useState([]);
+    const [playlists, setPlaylists] = useState([]);
+
+    useEffect(() => {
+        fetchProfile(accessToken).then((res) => setUserEmail(res.email));
+        fetchProfile(accessToken).then((res) => setProfileName(res.display_name));
+        fetchProfile(accessToken).then((res) => setUserImage(res.images?.[1]?.url));
```

Comment:
Unless I'm missing something:
Here we are calling the same api request, getting the same response, but just accessing a different property of the response object. This is being done three times.

When dealing with APIs, this can be costly - both financially [for paid APIs] as well as network costs, for our users. It has the potential of making the application slow, 3x slower to be precise.

We might want to consider putting these `set` calls into the same API call/`then` statement.

Reply:
You are totally right, these repeated calls are unnecessary, I will ammend this as part of my next PR.